include/alpaka/mem/buf/BufOacc.hpp

```diff
@@ -0,0 +1,652 @@
+            {
+                static_assert(
+                    !std::is_const<TElem>::value,
+                    "The elem type of the buffer can not be const because the C++ Standard forbids containers of const elements!");
+                static_assert(
+                    !std::is_const<TIdx>::value,
+                    "The idx type of the buffer can not be const!");
+            private:
+                using Elem = TElem;
+                using Dim = TDim;
+                //-----------------------------------------------------------------------------
+                //! Calculate the pitches purely from the extents.
+                template<
+                    typename TExtent>
+                ALPAKA_FN_HOST static auto calculatePitchesFromExtents(
+                    TExtent const & extent)
+                -> Vec<TDim, TIdx>
+                {
+                    Vec<TDim, TIdx> pitchBytes(Vec<TDim, TIdx>::all(0));
+                    pitchBytes[TDim::value - 1u] = extent[TDim::value - 1u] * static_cast<TIdx>(sizeof(TElem));
+                    for(TIdx i = TDim::value - 1u; i > static_cast<TIdx>(0u); --i)
+                    {
+                        pitchBytes[i-1] = extent[i-1] * pitchBytes[i];
+                    }
+                    return pitchBytes;
+                }
+
+            public:
+                //-----------------------------------------------------------------------------
+                //! Constructor
+                template<
+                    typename TExtent>
+                ALPAKA_FN_HOST BufOaccImpl(
+                    DevOacc const & dev,
+                    TElem * const pMem,
+                    TExtent const & extent) :
+                        m_dev(dev),
+                        m_extentElements(extent::getExtentVecEnd<TDim>(extent)),
+                        m_pitchBytes(calculatePitchesFromExtents(m_extentElements)),
+                        m_pMem(pMem)
+                {
+                    ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
+
+                    static_assert(
+                        TDim::value == alpaka::Dim<TExtent>::value,
+                        "The dimensionality of TExtent and the dimensionality of the TDim template parameter have to be identical!");
+                    static_assert(
+                        std::is_same<TIdx, Idx<TExtent>>::value,
+                        "The idx type of TExtent and the TIdx template parameter have to be identical!");
+                }
+
+            public:
+                DevOacc m_dev;
+                Vec<TDim, TIdx> m_extentElements;
+                Vec<TDim, TIdx> m_pitchBytes;
+                TElem* m_pMem;
+
+                    BufOaccImpl(const BufOaccImpl&) = delete;
+                    BufOaccImpl(BufOaccImpl &&) = default;
+                BufOaccImpl& operator=(const BufOaccImpl&) = delete;
+                BufOaccImpl& operator=(BufOaccImpl&&) = default;
+                    ~BufOaccImpl()
```

Comment:
Why is there a difference in indentation here?

Reply:
Ctors. Wrong style, fixed.